### PR TITLE
Added product module

### PR DESCRIPTION
**Pull request by manojkrsaini09**
product related functionality added

### CODE REVIEW

File: src/main/java/creativei/entity/Product.java [hunk 1]
```diff
@@ -22,10 +22,10 @@ public Product(Long id) {
         super(id);
     }
 
-    @Column(nullable = false)
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/Product.java]: Refer Schema: https://docs.zoho.com/sheet/open/6dl8ode3fdc2e90a54ff2b166b5011526ac37/sheets/Tables/ranges/E25
and update entity.

**Comment by manojkrsaini09** [comment]
@aayushKhandpur  I have added new fields please check now.

File: src/main/java/creativei/controller/ProductController.java [hunk 2]
```diff
@@ -94,4 +93,12 @@ ResponseObject createProduct(@RequestBody String productStr, HttpServletRequest
 			return ResponseObject.getResponse(ExceptionType.GENERAL_ERROR.getMessage(), ExceptionType.GENERAL_ERROR.getCode());
 		}
 	}
+
+	@GetMapping("/product-types")
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/ProductController.java]: use standard url all product url should start with /product
This one could be /product/type

File: src/main/java/creativei/entity/Product.java [hunk 3]
```diff
@@ -4,10 +4,12 @@
 
 import javax.persistence.Column;
 import javax.persistence.Entity;
+import javax.persistence.Index;
 import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
 
 @Entity
-@Table(name="Product")
+@Table(name="Product",indexes = @Index(columnList = "name,code"),uniqueConstraints=@UniqueConstraint(columnNames= {"code"}))
```
Comment (aayushKhandpur) [inline src/main/java/creativei/entity/Product.java]: Avoid adding unique constraint for now. We still need more clarity on Product code. We can avoid adding this constarint and should add later when confirmed

File: src/main/java/creativei/enums/ProductType.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	   for(ProductType type:ProductType.values()) {
+		   valueMap = new HashMap<>();
+		   valueMap.put("value", type.getType());
+		   valueMap.put("displayValue", type.getDisplayValue());
+		   valueMapList.add(valueMap);
+	   }
+	   return valueMapList;
+   }
+   
+   public static String getDisplayValueFromType(String type) {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/enums/ProductType.java]: Can we avoid this method? Refer CreativI Inquiry (LMS) project to find how we can define Enums

File: src/main/java/creativei/vo/ProductVO.java [hunk 5]
```diff
@@ -1,18 +1,25 @@
...
 import creativei.entity.Product;
+import creativei.enums.ProductType;
 
 public class ProductVO {
     Long id;
     String name;
     String code;
+    String brand;
+    String type;
+    String productTypeDisplayValue;
```
Comment (aayushKhandpur) [inline src/main/java/creativei/vo/ProductVO.java]: Why do we need both type and Display value?

File: src/main/resources/public/module/services/product-data-service.js [hunk 6]
```diff
@@ -22,5 +22,9 @@ function ProductDataService($http){
 		});
 	};
 	
+	ProductData.getProductTypes = function(){
+		return $http.get('/product-types');
```
Comment (aayushKhandpur) [inline src/main/resources/public/module/services/product-data-service.js]: Use url in a standardise way. All product service routes should be /product

File: src/main/java/creativei/entity/Product.java (hunk 3 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/entity/Product.java]: ok

File: src/main/java/creativei/enums/ProductType.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+		this.type = type;
+	}
+	public String getDisplayValue() {
+		return displayValue;
+	}
+	public void setDisplayValue(String displayValue) {
+		this.displayValue = displayValue;
+	}
+
+   public static List<Map<String,String>>  getAllProductType() {
```
[the remark this answers is not in the document]
Comment (manojkrsaini09) [reply src/main/java/creativei/enums/ProductType.java]: changed

File: src/main/java/creativei/enums/ProductType.java (hunk 4 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/enums/ProductType.java]: changed

File: src/main/java/creativei/vo/ProductVO.java (hunk 5 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/vo/ProductVO.java]: Type is actual string value going to save in db which will be useful in create/update call. Display value is useful in detail page.

File: src/main/java/creativei/service/ProductService.java [hunk 8]
```diff
@@ -9,5 +9,5 @@
 public interface ProductService {
     Product getById(Long id) throws ApplicationLevelException;
     List<Product> getAll() throws ApplicationLevelException;
-    Product create(ProductVO productVO) throws ApplicationLevelException;
+    Product create(Product product) throws ApplicationLevelException;
```
Comment (aayushKhandpur) [inline src/main/java/creativei/service/ProductService.java]: Use to approach as discussed. Response will be entity

File: src/main/resources/public/application.js [hunk 9]
```diff
@@ -33,7 +33,7 @@ app.config(['$stateProvider', '$urlRouterProvider', '$locationProvider', '$httpP
 		}
 	})
 	.state('Customer-Edit', {
-		url: '/create-customer',
+		url: '/create-customer/:customerId',
```
Comment (aayushKhandpur) [inline src/main/resources/public/application.js]: Url will be /customer/update/:customerId

File: src/main/resources/public/application.js [hunk 10]
```diff
@@ -46,7 +46,7 @@ app.config(['$stateProvider', '$urlRouterProvider', '$locationProvider', '$httpP
 		}
 	})
 	.state('Customer-Detail', {
-		url: '/customer-detail',
```
Comment (aayushKhandpur) [inline src/main/resources/public/application.js]: /customer/detail/:customerId

Comment (manojkrsaini09) [reply src/main/resources/public/application.js]: done

**Comment by manojkrsaini09** [comment]
@aayushKhandpur I have made changes suggested by you.

**Pull request merged**